In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv(r'Data/scopus_bd.csv', sep=",")

C:\Users\ECSIM\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2785: DtypeWarning: Columns (20,22) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [3]:
df.head(3)

Authors  \
0                                     Vanzolini P.E.   
1                                  De Wagener A.L.R.   
2  Mexias A.S., Berger G., Gomes M.E.B., Formoso ...   

                                        Author(s) ID  \
0                                        6602373241;   
1                                        7004422261;   
2  11739855600;7202349172;11738879800;6603689065;...   

                                               Title  Year  \
0  On Gymnodactylus amarali Barbour, 1925, with t...  2005   
1  Constraints to the implementation of effective...  2005   
2  Geochemical modeling of gold precipitation con...  2005   

                               Source title Volume Issue Art. No. Page start  \
0  Anais da Academia Brasileira de Ciencias     77     4      NaN        595   
1  Anais da Academia Brasileira de Ciencias     77     4      NaN        613   
2  Anais da Academia Brasileira de Ciencias     77     4      NaN        717   

  Page end         ...          \
0      611         ...           
1      623         ...           
2      728         ...           

                                      Index Keywords     ISSN ISBN CODEN  \
0  animal; article; Brazil; classification; compa...  13765.0  NaN   NaN   
1  article; developing country; environmental mon...  13765.0  NaN   NaN   
2                                     article; Butia  13765.0  NaN   NaN   

     PubMed ID Document Type Publication Stage Access Type  Source  \
0  1.63414e+07       Article             Final         NaN  Scopus   
1  1.63414e+07       Article             Final         NaN  Scopus   
2  1.63414e+07       Article             Final         NaN  Scopus   

                  EID  
0  2-s2.0-31044450573  
1  2-s2.0-31044447222  
2  2-s2.0-31044441319  

[3 rows x 28 columns]

#### Revisión rápida de columnas faltantes

In [6]:
df[['Page count','Cited by', 'DOI','Link', 'Affiliations','Authors with affiliations','Author Keywords', 'Index Keywords']][35:44]

Page count  Cited by                              DOI  \
35         NaN      12.0                              NaN   
36         NaN       1.0  10.1590/S0001-37652005000300001   
37         NaN      35.0  10.1590/S0001-37652005000300006   
38         NaN      40.0  10.1590/S0001-37652005000300005   
39         NaN       NaN  10.1590/S0001-37652005000300002   
40         NaN      45.0  10.1590/S0001-37652005000300004   
41         NaN       4.0  10.1590/S0001-37652005000300011   
42         NaN      33.0  10.1590/S0001-37652005000200004   
43         NaN      16.0  10.1590/S0001-37652005000200008   

                                                 Link  \
35  https://www.scopus.com/inward/record.uri?eid=2...   
36  https://www.scopus.com/inward/record.uri?eid=2...   
37  https://www.scopus.com/inward/record.uri?eid=2...   
38  https://www.scopus.com/inward/record.uri?eid=2...   
39  https://www.scopus.com/inward/record.uri?eid=2...   
40  https://www.scopus.com/inward/record.uri?eid=2...   
41  https://www.scopus.com/inward/record.uri?eid=2...   
42  https://www.scopus.com/inward/record.uri?eid=2...   
43  https://www.scopus.com/inward/record.uri?eid=2...   

                                         Affiliations  \
35  Departamento de Mineralogia e Geotectônica, In...   
36  Instituto de Matemática Pura e Aplicada, Estra...   
37  Departamento de Entomologia, Instituto Oswaldo...   
38  Instituto de Bioquímica Médica, Centro de Ciên...   
39  Departamento de Biologia, Fundação Oswalde Cru...   
40  Department of Biochemistry and Molecular Biolo...   
41  Departamento de Cartografia, FCT, UNESP, Rua R...   
42  Sección Limnología, Facultad de Ciencias, Univ...   
43  Laboratório de Imunologia Tumoral, Instituto d...   

                            Authors with affiliations  \
35  Gualda, G.A.R., Departamento de Mineralogia e ...   
36  Pereira, J.V., Instituto de Matemática Pura e ...   
37  Noireau, F., Departamento de Entomologia, Inst...   
38  Atella, G.C., Instituto de Bioquímica Médica, ...   
39  Santos, C.P., Departamento de Biologia, Fundaç...   
40  Azambuja, P., Department of Biochemistry and M...   
41  Dal-Poz, A.P., Departamento de Cartografia, FC...   
42  Brugnoli, E., Sección Limnología, Facultad de ...   
43  Esteves, M.B., Laboratório de Imunologia Tumor...   

                                      Author Keywords  \
35       Amphiboles; EPMA; Ferric iron; Stoichiometry   
36          Holomorphic Foliations; Jouanolou Theorem   
37  Chagas disease; Ecology; Silvatic environment;...   
38  Embryogenesis; Oogenesis; Rhodnius prolixus; T...   
39                                                NaN   
40  Parasites; Rhodnius prolixus; Trypanosomes; Ve...   
41  Digital photogrammetry; Road extraction; Road ...   
42  Biogeography; Invasive mollusk; Salinity; Uruguay   
43  Apoptosis; Lymphocytes; Membrane depolarizatio...   

                                       Index Keywords  
35                                                NaN  
36                                                NaN  
37                                        Triatominae  
38  Arthropoda; Hexapoda; Insecta; Rhodnius; Rhodn...  
39                                                NaN  
40  Insecta; Invertebrata; Reduviidae; Rhodnius; R...  
41                                                NaN  
42  Bivalvia; Limnoperna; Limnoperna fortunei; Luc...  
43                                Strophanthus gratus

Diferencia entre __'Affiliation'__ y __'Author with Affiliation'__?

¿Qué es Tipo de Acceso?

In [8]:
df['Access Type'].unique()

array([nan, 'Open Access', 'Final'], dtype=object)

¿Qué significa qué el tipo de acceso sea *Final*?
¿Será que Final es que se debe pagar para tener acceso al paper?
__Opinion Yorley Palacios__: El tipo de acceso puede tener un efecto en el potencial de citación, pues si es Open Acces más fácil se de a conocer el paper

¿Qué es CODEN?

In [10]:
df['CODEN'].unique()

array([nan, 'BTAPE', '268.0', '235.0', 'REAGD', 'ITECF', 'APSLA', 'JPKMA',
       'BEITD', 'CAPEE', 'PFECD', 'PCONF', '236.0', 'JSAEE', 'JSAMA',
       'RFIVF', '00319', 'RTFZD', 'TSAEA', 'SAJSA', 'WASAD'], dtype=object)

¿Cuales son los posibles estados de publicación?

In [11]:
df['Publication Stage'].unique()

array(['Final', 'Article in Press', 'Article'], dtype=object)

In [12]:
df[['Publication Stage','Cited by']].groupby(by='Publication Stage').count()

Cited by
Publication Stage          
Article                   0
Article in Press          0
Final                 32595

Cómo es de esperar no hay citaciones en papers sin publicar

In [14]:
df[['Publication Stage','Authors']].groupby(by='Publication Stage').count()

Authors
Publication Stage         
Article                  1
Article in Press         5
Final                51996

Solo se deben tener en cuenta los articulos en Fase Final, es decir ya publicados, sin embargo, este cambio no representa cambios significativos en los datos

In [17]:
len(df['EID'].unique())

51958

In [18]:
len(df['DOI'].unique())

26426

In [22]:
len(df['ISSN'].unique())

183

El ISSN (International Standard Serial Number, Número Internacional Normalizado de Publicaciones Seriadas) debería ser único

In [28]:
df[['ISSN','Authors']].groupby(by='ISSN').count().sort_values(by='Authors', ascending=False).head(n=10)

Authors
ISSN             
3781844      2469
7168756      2427
15168913     2331
16785878     2307
15161439     2274
3784738      1993
0376723X     1958
13765        1948
382353.0     1718
1046632      1648

In [29]:
df['Source'].unique()

array(['Scopus', 'Open Access'], dtype=object)

In [30]:
len(df['Source'].unique())

2

In [31]:
df[['Source','Authors']].groupby(by='Source').count()

Authors
Source              
Open Access        1
Scopus         52001

In [4]:
df.describe()

Year  Page count      Cited by
count  52002.000000  331.000000  32595.000000
mean    2010.456463   12.356495      7.172051
std        7.717393   17.646524     13.656950
min     1945.000000    1.000000      1.000000
25%     2007.000000    8.000000      2.000000
50%     2013.000000   10.000000      3.000000
75%     2016.000000   13.500000      8.000000
max     2019.000000  257.000000    414.000000

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 52002 entries, 0 to 52001
Data columns (total 28 columns):
Authors                      52002 non-null object
Author(s) ID                 52002 non-null object
Title                        52002 non-null object
Year                         52002 non-null int64
Source title                 52002 non-null object
Volume                       50956 non-null object
Issue                        50618 non-null object
Art. No.                     3184 non-null object
Page start                   47527 non-null object
Page end                     46564 non-null object
Page count                   331 non-null float64
Cited by                     32595 non-null float64
DOI                          26533 non-null object
Link                         52002 non-null object
Affiliations                 50159 non-null object
Authors with affiliations    51612 non-null object
Abstract                     52002 non-null object
Author Keywords           

Solo hay 10 Columnas que no contienen valores nulos.

Del Listado anterior de columnas, las que inicialmente se pueden considerar de utilidad para el estudio son:
__Author(s) ID, Year, Cited by, Source (Journal), Index KEyword, Author Keywords, page count( estimar para los papers que contengan Page start and page end), Abstract__

# Revisión Citaciones

Tenemos Datos Nulos?

In [53]:
df['Cited by'].isnull().any()

True

¿Cuántas Citaciones Nulas tenemos?
R: En total son 19.407 Citaciones sin valor

In [58]:
df['Cited by'][df['Cited by'].isnull()==True].shape

(19407,)

¿Una citación nula es igual a no citación, es decir un valor de 0?

In [64]:
df['Cited by'][df['Cited by']==0].shape

(0,)

Un documento sin Citación aparece como NaN, lo que implica que se deben actualizar estos datos, para que se representen con 0

# Revisión Abstracts

Tenemos Abstracts nulos?

In [28]:
df['Abstract'].isnull().any()

False

In [27]:
df['Abstract'][100:120]

100                              [No abstract available]
101    The White-browed Warbler Basileuterus leucoble...
102    The regional dialects or regiolects of the Ruf...
103    Research into the automated identification of ...
104    The habitat, but also the nest hole of a hole-...
105    Most territorial songbirds live in communicati...
106    Vocal signals play a very important role in th...
107    The song of the Brazilian population of the Hu...
108    A bird needs to keep track not only of social ...
109    Songbirds are well known for settling their di...
110    Guinea Pig Cavia porcellus pups emit high-pitc...
111    In this paper we prove a tangency principle (s...
112    Silver supported catalysts on alpha-alumina we...
113    Dockyards and harbors are recognized as being ...
114    We search for an amplification mechanism of th...
115    We investigate the influence of brain serotoni...
116    Fossil plants belonging to the morphogenera Gl...
117    Cell death by apoptosis 

Filas sin Abstracts se reportan como [No abstract available], entonces cuántos datos hay sin abstract

In [30]:
df.Abstract[df['Abstract']=='[No abstract available]'].count()

2967

Que porcentage del total de datos representan aquellos que no tienten Abstracts?

In [35]:
(df.Abstract[df['Abstract']=='[No abstract available]'].count())*100/df.shape[0]

5.705549786546671

¿Estos Artículos sin Abstract habrán sido citados?


In [51]:
df[['Abstract','Document Type','Cited by']][df['Abstract']=='[No abstract available]'].dropna().sort_values(by='Cited by', ascending=False)

Abstract     Document Type  Cited by
51825  [No abstract available]           Article     336.0
20648  [No abstract available]           Article     156.0
20680  [No abstract available]           Article     108.0
46488  [No abstract available]           Article     102.0
46593  [No abstract available]           Article      95.0
46530  [No abstract available]           Article      92.0
46109  [No abstract available]           Article      90.0
51848  [No abstract available]           Article      78.0
47288  [No abstract available]           Article      74.0
20709  [No abstract available]           Article      68.0
20896  [No abstract available]            Review      67.0
49888  [No abstract available]           Article      66.0
761    [No abstract available]           Article      65.0
45895  [No abstract available]            Review      61.0
20637  [No abstract available]            Review      60.0
45976  [No abstract available]           Article      59.0
20665  [No abstract available]           Article      55.0
51835  [No abstract available]           Article      55.0
46594  [No abstract available]           Article      55.0
49834  [No abstract available]           Article      54.0
20711  [No abstract available]           Article      54.0
46258  [No abstract available]  Conference Paper      54.0
51806  [No abstract available]            Review      53.0
648    [No abstract available]            Review      52.0
45823  [No abstract available]           Article      52.0
49889  [No abstract available]           Article      49.0
51849  [No abstract available]           Article      48.0
640    [No abstract available]           Article      47.0
49864  [No abstract available]           Article      46.0
51833  [No abstract available]           Article      42.0
...                        ...               ...       ...
47273  [No abstract available]           Article       1.0
47280  [No abstract available]           Article       1.0
20962  [No abstract available]            Review       1.0
20963  [No abstract available]           Article       1.0
21044  [No abstract available]         Editorial       1.0
21142  [No abstract available]         Editorial       1.0
47972  [No abstract available]         Editorial       1.0
47962  [No abstract available]              Note       1.0
47960  [No abstract available]      Short Survey       1.0
47958  [No abstract available]         Editorial       1.0
21364  [No abstract available]         Editorial       1.0
20934  [No abstract available]           Article       1.0
47259  [No abstract available]           Article       1.0
20914  [No abstract available]           Article       1.0
44812  [No abstract available]         Editorial       1.0
47116  [No abstract available]           Article       1.0
45757  [No abstract available]            Letter       1.0
20797  [No abstract available]           Article       1.0
48098  [No abstract available]              Note       1.0
45421  [No abstract available]         Editorial       1.0
44833  [No abstract available]           Article       1.0
20870  [No abstract available]           Article       1.0
43895  [No abstract available]         Editorial       1.0
20873  [No abstract available]           Article       1.0
47253  [No abstract available]           Article       1.0
20885  [No abstract available]           Article       1.0
48036  [No abstract available]      Short Survey       1.0
44772  [No abstract available]         Editorial       1.0
20909  [No abstract available]           Article       1.0
46481  [No abstract available]      Short Survey       1.0

[987 rows x 3 columns]

Cuántas de estos registros son de hecho Artículos
R:// 546.

__Opinion Yorley Palacios__ : Del total de datos sin abstracts se pueden eliminar 2967 - 546 = 2421

In [52]:
df[['Abstract','Document Type','Cited by']][df['Abstract']=='[No abstract available]'].dropna().sort_values(by='Cited by', ascending=False).groupby(by='Document Type').count()

Abstract  Cited by
Document Type                       
Article                546       546
Conference Paper        25        25
Editorial              112       112
Letter                  44        44
Note                   124       124
Review                  71        71
Short Survey            65        65

# Revisión Document Type

Tenemos Document Type Nulos

In [67]:
df["Document Type"].isnull().any()

True

In [ ]:
¿Cuántos Datos nulos tenemos?

In [72]:
df["Document Type"][df["Document Type"].isnull()==True].shape

(48,)

48 no es un número representativo :)

¿Cómo se distribuyen los tipos de documentos?
Nota: Se usa la columna Abstract porque no tiene valores nulos

In [80]:
df[["Document Type","Abstract"]].groupby(by='Document Type').count()

Abstract
Document Type              
Article               46917
Article in Press          1
Book Chapter              5
Conference Paper       1173
Conference Review         1
Editorial              1210
Erratum                 118
Letter                   95
Note                    359
Retracted                 3
Review                 1812
Short Survey            260

¿Cuáles son los Articulos más Citados?

In [76]:
df[df["Document Type"]=='Article'][['Document Type', 'Cited by']].sort_values(by='Cited by', ascending=False).dropna()

Document Type  Cited by
51269       Article     352.0
138         Article     351.0
14270       Article     343.0
51825       Article     336.0
9039        Article     334.0
51492       Article     302.0
46934       Article     294.0
51679       Article     249.0
8877        Article     235.0
14251       Article     229.0
26722       Article     221.0
51858       Article     221.0
46292       Article     201.0
51473       Article     192.0
49841       Article     188.0
51362       Article     187.0
14248       Article     186.0
51690       Article     181.0
51584       Article     175.0
30950       Article     159.0
20648       Article     156.0
49141       Article     154.0
2768        Article     151.0
290         Article     151.0
51625       Article     146.0
51357       Article     143.0
51148       Article     142.0
47222       Article     140.0
51555       Article     140.0
219         Article     140.0
...             ...       ...
3124        Article       1.0
11656       Article       1.0
24889       Article       1.0
20581       Article       1.0
12915       Article       1.0
3694        Article       1.0
20587       Article       1.0
3676        Article       1.0
20589       Article       1.0
2492        Article       1.0
20596       Article       1.0
11623       Article       1.0
12913       Article       1.0
8108        Article       1.0
3670        Article       1.0
3116        Article       1.0
41273       Article       1.0
11635       Article       1.0
20617       Article       1.0
12906       Article       1.0
2487        Article       1.0
41266       Article       1.0
41265       Article       1.0
41264       Article       1.0
41263       Article       1.0
11636       Article       1.0
11644       Article       1.0
11655       Article       1.0
20627       Article       1.0
39258       Article       1.0

[29997 rows x 2 columns]

In [8]:
df[df['Document Type']=='Article'].groupby(by='Year').count()

Authors  Author(s) ID  Title  Source title  Volume  Issue  Art. No.  \
Year                                                                        
1945        4             4      4             4       4      0         0   
1946        1             1      1             1       1      0         0   
1949        1             1      1             1       1      1         0   
1969        1             1      1             1       1      1         0   
1970        8             8      8             8       8      8         0   
1971       12            12     12            12      12     12         0   
1972        9             9      9             9       9      9         0   
1973       92            92     92            92      92     71         0   
1974       88            88     88            88      88     88         0   
1975       32            32     32            32      32     32         0   
1976       34            34     34            34      34     30         0   
1977       27            27     27            27      27     27         0   
1978       52            52     52            52      52     51         0   
1979       67            67     67            67      66     66         0   
1980       73            73     73            73      73     73         0   
1981       65            65     65            65      65     64         0   
1982       78            78     78            78      78     76         0   
1983       70            70     70            70      70     70         0   
1984       99            99     99            99      99     99         0   
1985      104           104    104           104     104    100         0   
1986       99            99     99            99      99     99         0   
1987      111           111    111           111     111    111         0   
1988       90            90     90            90      90     90         0   
1989       92            92     92            92      92     92         0   
1990       96            96     96            96      96     96         0   
1991       60            60     60            60      60     60         0   
1992       61            61     61            61      61     61         0   
1993       90            90     90            90      90     90         0   
1994       98            98     98            98      98     98         0   
1995      198           198    198           198     198    197         0   
1996      451           451    451           451     451    451         0   
1997      509           509    509           509     509    509         0   
1998      559           559    559           559     559    559         0   
1999      672           672    672           672     672    672         0   
2000      624           624    624           624     624    624         0   
2001      781           781    781           781     781    781         0   
2002      778           778    778           778     778    778         0   
2003      754           754    754           754     754    754         0   
2004      807           807    807           807     807    807         0   
2005      867           867    867           867     867    867         0   
2006     1299          1299   1299          1299    1299   1299         0   
2007     1422          1422   1422          1422    1422   1422         0   
2008     1807          1807   1807          1807    1753   1807         0   
2009     2085          2085   2085          2085    2008   2077         8   
2010     2468          2468   2468          2468    2355   2457        20   
2011     2365          2365   2365          2365    2232   2361        21   
2012     2541          2541   2541          2541    2445   2536        24   
2013     2906          2906   2906          2906    2829   2892        35   
2014     3331          3331   3331          3331    3223   3268        56   
2015     3222          3222   3222          3222    3183   3172        90   
20

# Revisión de Autores

Tenemos Varios Autores por Celda, como se muestra a continuación

In [73]:
a = df['Author(s) ID'][2]
a

'11739855600;7202349172;11738879800;6603689065;6507369527;11739533300;11739290800;'

Uno a uno se separan así:

In [10]:
a.split(sep=";")

['11739855600',
 '7202349172',
 '11738879800',
 '6603689065',
 '6507369527',
 '11739533300',
 '11739290800',
 '']

## Citación de Revistas por año

In [21]:
df[df['Document Type']=='Article'].pivot_table( index=('Year'), columns=('Source title'), values='Cited by', aggfunc=np.sum, fill_value=0)

Source title  ANAIS DA ACAD. BRASILEIRA DE CIENCIAS.  \
Year                                                   
1945                                               0   
1946                                               0   
1949                                               0   
1969                                               0   
1970                                               0   
1971                                               0   
1972                                               0   
1973                                               0   
1974                                               0   
1975                                               0   
1976                                               0   
1977                                               0   
1978                                               0   
1979                                               0   
1980                                               0   
1981                                               0   
1982                                               0   
1983                                               0   
1984                                               0   
1985                                               0   
1986                                               0   
1987                                               0   
1988                                               0   
1989                                               0   
1990                                               0   
1991                                               0   
1992                                               0   
1993                                               0   
1994                                               0   
1995                                               0   
1996                                               0   
1997                                               0   
1998                                               0   
1999                                               0   
2000                                               0   
2001                                               0   
2002                                               0   
2003                                               0   
2004                                               0   
2005                                               0   
2006                                               0   
2007                                               0   
2008                                               0   
2009                                               0   
2010                                               0   
2011                                               0   
2012                                               0   
2013                                               0   
2014                                               0   
2015                                               0   
2016                                               0   
2017                                               0   
2018                                               0   
2019                                               0   

Source title  Acta Colombiana de Psicologia  \
Year                                          
1945                                      0   
1946                                      0   
1949                                      0   
1969                                      0   
1970                                      0   
1971                                      0   
1972                                      0   
1973                                      0   
1974                                      0   
1975                                      0   
1976                                      0   
1977                                      0   
1978                                      0   
1979                                      0   
1980                                      0   
1981                                      0   
1982             